Fix IPv6 missing brackets in httpProxy.ServerOptions

Dominic1DL opened the pull request:
This pull request should fix: #5743
Signed-off-by: dominic_dl <me@dominic.link>

Dominic1DL committed:
Fix IPv6 missing brackets in httpProxy.ServerOptions

Signed-off-by: dominic_dl <me@dominic.link>

ixrock reviewed (dismissed):
Haven't checked if the fix is really working. 
Otherwise code LGTM.

Dominic1DL committed:
Minor changes

- fix lint error
- move the code inside the clusterUrl.hostname check
- add small Comment to explane the code

Signed-off-by: dominic_dl <me@dominic.link>
